src/ViewModel/Select.php

```diff
@@ -35,11 +36,16 @@ public function __construct(
         bool $required = null,
         bool $disabled = null,
         string $state = null,
-        string $message = null
+        MessageGroup $messageGroup = null
     ) {
         Assertion::notEmpty($options);
         Assertion::allIsInstanceOf($options, SelectOption::class);
-        Assertion::nullOrChoice($state, [self::STATE_ERROR, self::STATE_VALID]);
+        Assertion::nullOrChoice($state, [self::STATE_INVALID, self::STATE_VALID]);
+        if ($state === self::STATE_INVALID) {
+            if (is_null($messageGroup) || empty($messageGroup['errorText'])) {
+                throw new InvalidArgumentException('There must be a message group containing error text if the state is error.');
```

Comment:
Could use `Assertion::notNull($messageGroup)` and `Assertion::notBlank($messageGroup['errorText'])`.